src/pages/_middleware.tsx

```diff
@@ -0,0 +1,43 @@
+import { NextResponse } from "next/server";
+
+export function middleware() {
```

Comment:
This seems useful enough to be broken out as a dependency, i.e. `next-opinionated-middleware`